pkg/env/armhelper.go

```diff
@@ -113,23 +121,26 @@ func newARMHelper(ctx context.Context, log *logrus.Entry, env Interface) (ARMHel
 		log: log,
 		env: env,
 
+		fpGraphClient:   fpGraphClient,
 		roleassignments: authorization.NewRoleAssignmentsClient(env.Environment(), env.SubscriptionID(), armAuthorizer),
-		applications:    graphrbac.NewApplicationsClient(env.Environment(), env.TenantID(), fpGraphAuthorizer),
 	}, nil
 }
 
-func (ah *armHelper) EnsureARMResourceGroupRoleAssignment(ctx context.Context, fpAuthorizer autorest.Authorizer, resourceGroup string) error {
+func (ah *armHelper) EnsureARMResourceGroupRoleAssignment(ctx context.Context, resourceGroup string) error {
 	ah.log.Print("ensuring resource group role assignment")
 
-	res, err := ah.applications.GetServicePrincipalsIDByAppID(ctx, ah.env.FPClientID())
+	principalID, err := utilgraph.GetServicePrincipalIDByAppID(ctx, ah.fpGraphClient, ah.env.FPClientID())
 	if err != nil {
 		return err
 	}
+	if principalID == nil {
+		return fmt.Errorf("no service principal found for application ID '%s'", ah.env.FPClientID())
```

Comment:
again, this might be an eventual consistency error that's not caught on our `EnsureResourceGroup`.  This would only be replicated in development environments so it wouldn't hit prod clusters, but still something to have a workaround for.

Reply:
Would it be best to wait and see if this is going to be a problem, then revisit?

Reply:
Yes, giving adaquite time for it to propagate in lower environments before we roll it to prod.